File: common/src/main/java/com/hartwig/actin/treatment/datamodel/TrialIdentification.java
package com.hartwig.actin.treatment.datamodel;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class TrialIdentification {

    @NotNull
    public abstract String trialId();

    public abstract boolean open();

    @NotNull
    public abstract String acronym();

    @NotNull
    public abstract String title();
}

File: common/src/main/kotlin/com/hartwig/actin/trial/datamodel/TrialIdentification.kt
package com.hartwig.actin.trial.datamodel

data class TrialIdentification(
    val trialId: String,
    val open: Boolean,
    val acronym: String,
    val title: String,
    val nctId: String? = null,
    val phase: TrialPhase? = null,
    val mainTumorTypes: Set<String>?
)
